core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -363,6 +364,7 @@ public void testSegments() throws Exception {
             assertThat(segments.get(1).isCompound(), equalTo(false));
 
             engine.config().setCompoundOnFlush(true);
+            engine.onSettingsChanged();
```

Comment:
> So it's not strictly speaking needed, but you rather not rely on the refresh. 

It is needed, otherwise the test fails, with my changes here, because I removed the "secretly push settings changes to IW on refresh".